netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -595,17 +611,23 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
       NettyClientHandler.this.onRstStreamRead(streamId, errorCode);
     }
 
-    @Override public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data)
-        throws Http2Exception {
+    @Override
+    public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2Exception {
       Http2Ping p = ping;
-      if (p != null) {
+      if (data.readLong() == flowControlPing.payload()) {
+        data.readerIndex(0);
+        flowControlPing.incrementDataSincePing(data.readableBytes());
+        flowControlPing.updateWindow();
+        logger.log(Level.FINE, String.format("OBDP: {0}", flowControlPing.getDataSincePing()));
```

Comment:
`logger.log(Level.FINE, "OBDP: {0}", flowControlPing.getDataSincePing());`

That 1) will format correctly (`String.format()` uses printf-style formatting) and 2) only format when logging is taking place.